prow/cmd/generic-autobumper/bumper/bumper.go

```diff
@@ -412,6 +422,132 @@ func Run(o *Options) error {
 	return nil
 }
 
+// Run2 is the entrypoint which will update Prow config files based on the
+// provided options.
+//
+// updateFunc: a function that returns commit message and error
+func Run2(o *Options, prh PRHandler) error {
+	if err := validateOptions(o); err != nil {
+		return fmt.Errorf("error validating options: %w", err)
+	}
+
+	var sa secret.Agent
+
+	stdout := HideSecretsWriter{Delegate: os.Stdout, Censor: &sa}
+	stderr := HideSecretsWriter{Delegate: os.Stderr, Censor: &sa}
+	if o.SkipPullRequest {
+		logrus.Debugf("--skip-pull-request is set to true, won't create a pull request.")
+	}
+	if o.Gerrit == nil {
+		if err := sa.Start([]string{o.GitHubToken}); err != nil {
+			return fmt.Errorf("failed to start secrets agent: %w", err)
+		}
+
+		gc := github.NewClient(sa.GetTokenGenerator(o.GitHubToken), sa.Censor, github.DefaultGraphQLEndpoint, github.DefaultAPIEndpoint)
+
+		// Make change, commit and push
+		for _, f := range prh.Changes() {
+			msg, err := f()
+			if err != nil {
+				return fmt.Errorf("failed to process provided function: %w", err)
+			}
+
+			changed, err := HasChanges()
+			if err != nil {
+				return fmt.Errorf("error occurred when checking changes: %w", err)
+			}
+
+			if !changed {
+				// TODO(chaodaiG): decide how to handle this with multiple commits
+				logrus.Info("Nothing changed, exiting ...")
+				return nil
+			}
+
+			if err := gitCommit(o.GitName, o.GitEmail, msg, stdout, stderr, false); err != nil {
+				return fmt.Errorf("failed git commit: %w", err)
+			}
+		}
+
+		if err := gitPush(fmt.Sprintf("git@github.com:%s/%s.git", o.GitHubLogin, o.RemoteName),
+			o.HeadBranchName, stdout, stderr, o.SkipPullRequest); err != nil {
+			return fmt.Errorf("failed to push changes to the remote branch: %w", err)
+		}
+
+		summary, body, err := prh.PRTitleBody()
+		if err != nil {
+			return fmt.Errorf("failed creating PR summary and body: %w", err)
+		}
+		if o.GitHubBaseBranch == "" {
+			repo, err := gc.GetRepo(o.GitHubOrg, o.GitHubRepo)
+			if err != nil {
+				return fmt.Errorf("failed to detect default remote branch for %s/%s: %w", o.GitHubOrg, o.GitHubRepo, err)
+			}
+			o.GitHubBaseBranch = repo.DefaultBranch
+		}
+		if err := updatePRWithLabels2(gc, o.GitHubOrg, o.GitHubRepo,
+			getAssignment(o.AssignTo, o.OncallAddress, o.OncallGroup), o.GitHubLogin,
+			o.GitHubBaseBranch, o.HeadBranchName, updater.PreventMods, summary, body, o.Labels, o.SkipPullRequest); err != nil {
+			return fmt.Errorf("failed to create the PR: %w", err)
+		}
+	} else {
```

Comment:
Nit: Maybe factor the gerrit/non-gerrit handling out into distinct funcs? By the time I reached the `else` I had forgotten what the `if` was about :P

Reply:
Good idea, done